src/iinit.cpp

```diff
@@ -66,11 +69,225 @@ mkrodsdir() {
+        }
+        else {
+            fmt::print("failed to open environment file [{}]\n", env_file);
+        }
+    } // save_updates_to_irods_environment
+
+    auto option_specified(std::string_view _option, int argc, char** argv) -> bool
+    {
+        for (int arg = 0; arg < argc; ++arg) {
+            if (!argv[arg]) {
+                continue;
+            }
+
+            if (_option == argv[arg]) {
+                // parseCmdLineOpt is EVIL and requires this. Please don't ask why.
+                argv[arg] = "-Z";
+                return true;
+            }
+        }
+
+        return false;
+    } // option_specified
+
+    auto set_env_from_prompt(char _setting[], const char* _prompt, std::size_t _len) -> void
+    {
+        // If the setting already has a value, use that as the default value and show it in the prompt.
+        if (0 == std::strlen(_setting)) {
+            fmt::print("{}: ", _prompt);
+        }
+        else {
+            fmt::print(fmt::runtime(fmt::format("{} [{{}}]: ", _prompt)), _setting);
+        }
+
+        std::string response;
+        std::getline(std::cin, response);
+
+        // If the response is empty, don't copy anything. If there is a value in _setting already, it will be used
+        // as the default response. If there is nothing, then there is nothing to copy anyway.
+        if (!response.empty()) {
+            std::strncpy(_setting, response.c_str(), _len);
+        }
+    } // set_env_from_prompt
+
+    auto set_env_from_prompt(char _setting[], const char* _default, const char* _prompt, std::size_t _len) -> void
+    {
+        const bool env_has_value = 0 != std::strlen(_setting);
+        const char* default_value = env_has_value ? _setting : _default;
+
+        fmt::print(fmt::runtime(fmt::format("{} [{{}}]: ", _prompt)), default_value);
+        std::string response;
+        std::getline(std::cin, response);
+
+        // The else case indicates that the response was empty and the environment already has a value. If the
+        // environment has a value already, the default value is the existing value (meaning, there will be no
+        // change in the environment). Therefore, no copy occurs in the else case as no change will occur.
+        if (!response.empty()) {
+            std::strncpy(_setting, response.c_str(), _len);
+        }
+        else if (!env_has_value) {
+            std::strncpy(_setting, default_value, _len);
+        }
+    } // set_env_from_prompt
+
+    auto set_env_from_prompt(int& _setting, int _default, const char* _prompt) -> void
+    {
+        const bool env_has_value = 0 != _setting;
+        const auto default_value = env_has_value ? _setting : _default;
+
+        fmt::print(fmt::runtime(fmt::format("{} [{{}}]: ", _prompt)), default_value);
+        std::string response;
+        std::getline(std::cin, response);
+
+        // The else case indicates that the response was empty and the environment already has a value. If the
```

Comment:
same.  not quite sure this is clear.

Reply:
The comment is explaining why there is no else case here (it's a no-op). Perhaps it would be more clear if the comment (or a separate comment) appeared where the else case would be if there was one? Open to suggestions on alternative wording as well.

Reply:
It could be replaced with a table/list. Or, you can try inverting the explanation.

I don't feel it needs a comment. The code is small enough that it is pretty easy to understand.

Reply:
Seeing as the comments have generated more confusion than they have alleviated, I think it may be better to just remove it. I find the code clear, personally, but that's because I wrote it. :)